### PR TITLE
fix(entity-detail): fix field touched bug in DetailForm

### PR DESCRIPTION
**Pull request by khanlifts**

### CODE REVIEW

File: packages/entity-detail/src/components/DetailForm/DetailForm.js [hunk 1]
```diff
@@ -12,8 +12,10 @@ import readOnlyFormFieldMapping from '../../util/detailView/readOnlyFormFieldMap
 import StyledDetailForm from './StyledDetailForm'
 
 export class DetailForm extends React.Component {
-  componentWillReceiveProps(nextProps) {
-    this.props.fireTouched(nextProps.anyTouched)
+  componentDidUpdate(prevProps, prevState, snapshot) {
+    if (prevProps !== this.props) {
```
Comment (MethodenMann) [inline packages/entity-detail/src/components/DetailForm/DetailForm.js]: what is the point of this `if`?

File: packages/entity-detail/src/components/DetailForm/DetailForm.js [hunk 2]
```diff
@@ -12,8 +12,10 @@ import readOnlyFormFieldMapping from '../../util/detailView/readOnlyFormFieldMap
 import StyledDetailForm from './StyledDetailForm'
 
 export class DetailForm extends React.Component {
-  componentWillReceiveProps(nextProps) {
-    this.props.fireTouched(nextProps.anyTouched)
+  componentDidUpdate(prevProps, prevState, snapshot) {
```
Comment (MethodenMann) [inline packages/entity-detail/src/components/DetailForm/DetailForm.js]: snapshot can be omitted

File: packages/entity-detail/src/components/DetailForm/DetailForm.js [hunk 3]
```diff
@@ -128,6 +130,7 @@ DetailForm.propTypes = {
   anyTouched: PropTypes.bool,
   formErrors: PropTypes.object,
   valid: PropTypes.bool,
+  isDirty: PropTypes.bool,
```
Comment (MethodenMann) [inline packages/entity-detail/src/components/DetailForm/DetailForm.js]: the save button uses "anyTouched" as well. `isDirty` could there be more suitable as well. Remove any unused properties at the end.

File: packages/entity-detail/src/components/DetailForm/DetailForm.js (hunk 1 above)
Comment (khanlifts) [reply packages/entity-detail/src/components/DetailForm/DetailForm.js]: standard componendDidUpdate check to control the calls, prevents infinite loops as well.

File: packages/entity-detail/src/components/DetailForm/DetailForm.js [hunk 4]
```diff
@@ -91,9 +93,9 @@ export class DetailForm extends React.Component {
           />
           {!this.isReadOnlyForm()
           && <div>
-            {!props.valid && props.anyTouched && <ErrorBox formErrors={props.formErrors} showErrors={this.showErrors}/>}
+            {!props.valid && props.isDirty && <ErrorBox formErrors={props.formErrors} showErrors={this.showErrors}/>}
```
Comment (MethodenMann) [inline packages/entity-detail/src/components/DetailForm/DetailForm.js]: Not quite perfect. 
Reproduce problem:
In the mocked form touch, but not edit, the "Callname" Field. Save button is not disabled but error on field is shown.

image

**Pull request merged**